mtags/src/main/scala-3/scala/meta/internal/pc/InferredTypeProvider.scala

```diff
@@ -251,4 +255,32 @@ final class InferredTypeProvider(
     end match
   end inferredTypeEdits
 
+  private def namePos(text: String, tree: untpd.NameTree)(using
+      Context
+  ): SourcePosition =
+    val realName = tree.name.stripModuleClassSuffix.lastPart.toString.toList
+
+    def lookup(
+        idx: Int,
+        start: Option[(Int, List[Char])],
+        withBacktick: Boolean
+    ): SourcePosition =
+      start match
+        case Some((start, nextMatch :: left)) =>
+          if text.charAt(idx) == nextMatch then
+            lookup(idx + 1, Some((start, left)), withBacktick)
+          else lookup(idx + 1, None, withBacktick = false)
+        case Some((start, Nil)) =>
+          val end = if withBacktick then idx + 1 else idx
+          tree.source.atSpan(Span(start, end, start))
+        case None =>
+          val ch = text.charAt(idx)
+          if ch == realName.head then
```

Comment:
Maybe it's too cautious, but can we use `headOption` to avoid `NoSuchElementException` here?

Reply:
Is it possible to get zero length name?

Reply:
I don't think it's possible, we could add a require with a proper message just in case

Reply:
added a check for that